test/yjsTestEditor.ts

```diff
@@ -0,0 +1,152 @@
+import { Editor, Operation } from 'slate';
+import * as Y from 'yjs';
+import { applySlateOps as applySlateOperations } from '../src/apply';
+import { toSlateOps } from '../src/convert';
+import { SyncDoc, SyncElement } from '../src/model';
+
+export interface YjsTestEditor extends Editor {
+  isRemote: boolean;
+  doc: Y.Doc;
+  syncDoc: SyncDoc;
+  shouldCaptureYjsUpdates: boolean;
+  capturedYjsUpdates: Uint8Array[];
+  onChangeComplete: () => void;
+}
+
+export const YjsTestEditor = {
+  /**
+   * Apply slate ops to Yjs
+   */
+  applySlateOpsToYjs: (e: YjsTestEditor, operations: Operation[]) => {
+    e.doc.transact(() => {
+      applySlateOperations(e.syncDoc, operations);
+    });
+  },
+
+  /**
+   * Apply Yjs events to slate
+   */
+  applyYjsEventsToSlate: (e: YjsTestEditor, events: Y.YEvent[]) => {
+    const remoteEvents = events.filter((event) => !event.transaction.local);
+    if (remoteEvents.length == 0) {
+      return;
+    }
+
+    e.isRemote = true;
+
+    Editor.withoutNormalizing(e, () => {
+      toSlateOps(remoteEvents).forEach((op) => {
+        e.apply(op);
+      });
+    });
+
+    Promise.resolve().then(() => (e.isRemote = false));
```

Comment:
You duplicated a lot of code from withYjs and the yjs editor. If you write the YjsTestEditor to extend the Yjs editor, it's implementation would be tested as well and it would remove a lot of duplicate code.

Reply:
(And it then would probably make sense to rename withYjsTest to withText and the editor to TestEditor)

Reply:
Do you mean `withText` here or `withTest`?

Reply:
Sorry, I meant withTest